src/nvim/ops.c

```diff
@@ -2971,7 +2971,7 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
     y_array = reg->y_array;
   }
 
-  if (curbuf->terminal) {
+  if (y_size > 0 && curbuf->terminal) {
```

Comment:
Move the check for `y_size` to the start of `terminal_paste()`.

Now we would run all the code below although we're in a terminal.

Reply:
Done